apps/astarte_appengine_api/lib/astarte_appengine_api_web/controllers/fallback_controller.ex

```diff
@@ -205,6 +205,13 @@ defmodule Astarte.AppEngine.APIWeb.FallbackController do
     |> render(:"422_unexpected_object_key")
   end
 
+  def call(conn, {:error, :unset_not_allowed}) do
+    conn
+    |> put_status(:bad_request)
+    |> put_view(Astarte.AppEngine.APIWeb.ErrorView)
+    |> render(:"400_unset_not_allowed")
```

Comment:
I would argue that this is more akin to a 422 error, as
>  The request was well-formed but was unable to be followed due to semantic errors.

([source](https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Status#client_error_responses)) In the same fashion as we do for unexpected object keys or invalid device attributes.
Moreover, it would be great to document this new error in the [API docs](https://github.com/astarte-platform/astarte/blob/release-1.2/apps/astarte_appengine_api/priv/static/astarte_appengine_api.yaml)